plugins/csp-lod-bodies/src/TileSourceWebMapService.cpp

```diff
@@ -81,28 +81,39 @@ bool loadImpl(
     // two requests are made. For those, only half of the pixels contain valid data (above or below
     // the diagonal).
     int imagelength{};
-    TIFFGetField(data, TIFFTAG_IMAGELENGTH, &imagelength);
-    for (int y = 0; y < imagelength; y++) {
-      if (which == CopyPixels::eAll) {
-        TIFFReadScanline(data, &tile->data()[resolution * y], y);
-      } else if (which == CopyPixels::eAboveDiagonal) {
-        std::vector<float> tmp(resolution);
-        TIFFReadScanline(data, tmp.data(), y);
-        int offset = resolution * y;
-        int count  = resolution - y - 1;
-
-        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
-        std::memcpy(tile->data().data() + offset, tmp.data(), count * sizeof(float));
-      } else if (which == CopyPixels::eBelowDiagonal) {
-        std::vector<float> tmp(resolution);
-        TIFFReadScanline(data, tmp.data(), y);
-        int offset = resolution * y + (resolution - y);
-        int count  = y;
-
-        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
-        std::memcpy(
-            tile->data().data() + offset, tmp.data() + resolution - y, count * sizeof(float));
+    int tiffReturn{};
+    if (TIFFGetField(data, TIFFTAG_IMAGELENGTH, &imagelength)) {
```

Comment:
I think it would be more readable if we would return early like in the case above. You will need to remove the `else` below and fix the indentation. So do not directly accept this but rather change this on your end :) 

```suggestion
    if (TIFFGetField(data, TIFFTAG_IMAGELENGTH, &imagelength) == 0) {
      logger().debug("TIFFGetField failed: Removing invalid cache file '{}'.", *cacheFile);
      boost::filesystem::remove(*cacheFile);
      return false;
    }
```